client/src/components/mapping/FhirMappingPanel/FhirResourceTree/index.tsx

```diff
@@ -260,23 +251,37 @@ const FhirResourceTree = ({ onClickCallback }: Props) => {
     } else {
       if (selectedNode) selectedNode.isSelected = false;
       node.isSelected = true;
-      setSelectedNode(node);
+      setSelectedNode(node as TreeNode);
       onClickCallback(node.nodeData);
     }
     setNodes([...nodes]);
   };
 
+  const handleContextMenu = async (
+    node: ITreeNode<Attribute>
+  ): Promise<void> => {
+    const attribute = node.nodeData!;
+    if (
+      (attribute.isArray || !attribute.isPrimitive) &&
+      (!node.childNodes || node.childNodes.length === 0)
+    ) {
+      const children = await fetchAttributeDefinition(attribute.types[0]);
+      children.forEach(child => attribute.addChild(child));
+      node.childNodes = genTreeLevel(attribute.children);
+    }
+    setNodes([...nodes]);
+  };
+
   useEffect(() => {
     setNodes(nodes => {
-      for (const node of nodes) {
-        updateSecondaryLabel(node);
-      }
+      nodes.forEach(n => n.updateSecondaryLabel());
       return [...nodes];
     });
   }, [attributesForResource]);
 
   useEffect(() => {
     if (!loading) setNodes(genTreeLevel(fhirStructure));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This comment should be removed ?

Reply:
nope it can't: it prevents lint errors that we cannot avoid (we cannot have an exhaustive list of dependency array in this specific case).